src/VisualStudio/VisualBasic/Impl/Options/Formatting/CodeStylePage.vb

```diff
@@ -1,16 +1,171 @@
+            Return New GridOptionPreviewControl(serviceProvider, Function(o, s) New StyleViewModel(o, s), AddressOf GetCurrentEditorConfigOptionsString)
+        End Function
+
+        Friend Shared Function GetCurrentEditorConfigOptionsString(ByVal optionSet As OptionSet) As String
+            Dim editorconfig = New StringBuilder()
+
+            ' Core EditorConfig Options
+            editorconfig.AppendLine("###############################")
+            editorconfig.AppendLine("# Core EditorConfig Options   #")
+            editorconfig.AppendLine("###############################")
+            editorconfig.AppendLine("# You can uncomment the next line if this is your top-most .editorconfig file.")
+            editorconfig.AppendLine("# root = true")
+            editorconfig.AppendLine()
+            editorconfig.AppendLine("# Basic files")
+            editorconfig.AppendLine("[*.vb]")
+            editorconfig.Append("indent_style = ")
+            editorconfig.AppendLine(BasicCodeStyleOptions_GenerateEditorconfig(optionSet, FormattingOptions.UseTabs))
+            editorconfig.Append("indent_size = ")
+            editorconfig.AppendLine(optionSet.GetOption(FormattingOptions.IndentationSize, "csharp").ToString())
+            editorconfig.Append("insert_final_newline = ")
+            editorconfig.AppendLine(optionSet.GetOption(FormattingOptions.InsertFinalNewLine).ToString().ToLower())
+
+            editorconfig.AppendLine()
+            ' .NET Coding Conventions
+            editorconfig.AppendLine("###############################")
+            editorconfig.AppendLine("# .NET Coding Conventions     #")
+            editorconfig.AppendLine("###############################")
+            editorconfig.AppendLine("# Organize usings")
+            editorconfig.Append("dotnet_sort_system_directives_first = ")
+            editorconfig.AppendLine(optionSet.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, "csharp").ToString().ToLower())
+
+            editorconfig.AppendLine()
+            ' this. preferences
+            editorconfig.AppendLine("# this. preferences")
+            editorconfig.Append("dotnet_style_qualification_for_field = ")
+            editorconfig.AppendLine(BasicCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyFieldAccess))
+            editorconfig.Append("dotnet_style_qualification_for_property = ")
+            editorconfig.AppendLine(BasicCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyPropertyAccess))
+            editorconfig.Append("dotnet_style_qualification_for_method = ")
+            editorconfig.AppendLine(BasicCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyMethodAccess))
+            editorconfig.Append("dotnet_style_qualification_for_event = ")
+            editorconfig.AppendLine(BasicCodeStyleOptions_GenerateEditorconfig(optionSet, CodeStyleOptions.QualifyPropertyAccess))
```

Comment:
1. property -> event
2. It feels like there should be some helpers that can generate these lines regardless of C# or Vb.  then we can just unify a lot of code and simplify the amount of work it is to print al lthis out.